Libs/MRML/Core/vtkMRMLSliceNode.cxx

```diff
@@ -333,111 +326,229 @@ ::IsDisplayableInThreeDView(const char* viewNodeID)const
 }
 
 //----------------------------------------------------------------------------
-void vtkMRMLSliceNode::SetOrientation(const char* orientation)
+bool vtkMRMLSliceNode::SetOrientation(const char* orientation)
 {
   if (!orientation)
     {
-    return;
+    return false;
     }
- if (!strcmp(orientation, "Axial"))
+
+  vtkMatrix4x4 *orientationPreset;
+  orientationPreset = this->GetSliceOrientationPreset(orientation);
+
+  if(!strcmp(orientation, "Reformat"))
     {
-    this->SetOrientationToAxial();
+    this->SetOrientationString(orientation);
     }
-  else if (!strcmp(orientation, "Sagittal"))
+
+  if(orientationPreset == NULL)
+    {
+    return false;
+    }
+
+  //SetOrientationString() and SetOrientationReference()
+  //behaviour will be mantained for backward compatibility
+  this->SliceToRAS->DeepCopy(orientationPreset);
+  this->SetOrientationString(orientation);
+  this->SetOrientationReference(orientation);
+  this->UpdateMatrices();
+  return true;
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLSliceNode::SetOrientationToReformat()
+{
+    // Don't need to do anything.  Leave the matrices where they were
+    // so the reformat starts where you were.
+
+    this->SetOrientationString( "Reformat" );
```

Comment:
I think SetOrietationString is not really deprecated. It is a string that tell us which preset is using at the moment the matrix sliceToRAS